### ISSUE TITLE
GoogleSignIn - it says to register in Firebase, no Firebase used in example app

### ISSUE THREAD

**Issue by giaur500** [flutter/flutter#68039]
https://pub.dev/packages/google_sign_in

First step it says I need to register with Firebase:

`To access Google Sign-In, you'll need to make sure to register your application.` - register here: https://developers.google.com/mobile/add?platform=android
However, it's completly opposite from official GoogleSignIn native, for Android Google says I need to register completly different way (in Google Console, not Firebase). Your docs says I need to register in Firebase instead, but not download Google Services json, which makes no sense. Also, you never use anything Firebase related. Please clarify what do you mean by "register application", do I need to connect it to Firebase? Firebase login with Google is different thing, as I can see not covered with your plugin. 

I am unable to run your sample, as I docs seems to be confusing (or outdated) when saying about register app and I am confused where to start. Can please clarify how and where I need to register my app actually?

**Comment by iapicca** [comment]
there's a [medium article](https://medium.com/flutter-community/flutter-sign-in-with-google-in-android-without-firebase-a91b977d166f) on how to use [google_sign_in](https://pub.dev/packages/google_sign_in) w/o firebase
but I agree with @giaur500  , with the current documentation 
it might seems that the only officially supported approach is though firebase, which I believe is misleading

**Comment by giaur500** [comment]
Well I started digging and I figured out, that when you are registering app in Firebase, corresponding project seems to be added automatically to Google Cloud Console. So it may work as described in plugin readme. But I am not sure if it's still true if you only create Firebse app, but never actually integrate it to connect to Firebase. Looking into plugin source code, I see it does not use Firebase. And plugin readme does not mention I need to integrate with Firebase and connect app to Firebse. It only says to register app, which is unclear what it means actually
